### ISSUE TITLE
Build error in TypeScript 5.0

### ISSUE THREAD

**Issue by CatsMiaow** [iamolegga/nestjs-pino#1400]
<img width="999" alt="image" src="https://user-images.githubusercontent.com/1300172/227870175-c4afad80-61c1-4ae8-9169-cdcef17d4469.png">

> error TS1239: Unable to resolve signature of parameter decorator when called as an expression. Argument of type 'undefined' is not assignable to parameter of type 'string | symbol'.

Related issues
- https://github.com/nestjs/nest/pull/10970
- https://github.com/microsoft/TypeScript/pull/53365

`InjectPinoLogger` type output from `nestjs-pino@v3.1.2`
```ts
// AS-IS
export declare function InjectPinoLogger(context?: string): (target: object, key: string | symbol, index?: number | undefined) => void;
// TO-BE
export declare function InjectPinoLogger(context?: string): (target: object, key: string | symbol | undefined, index?: number | undefined) => void;
```

The above return type is created as the return type of the `Inject` function of `@nestjs/common`.
This issue has been fixed in `@nestjs/common@v9.3.0`.
In nestjs-pino, you need to build and deploy it again with the [updated internal dependency](https://github.com/iamolegga/nestjs-pino/pull/1393) version to change the type.

**Comment by mathieuletyrant** [comment]
I have the same issue on my projects, and it's blocking us to use TS 5, do you need help @iamolegga?

**Comment by iamolegga** [comment]
This is not just an update of the typescript version and fixing what's broken. This should also be tested with different typescript versions of dependent projects. So, it requires a bit more time to not break all the things. I should be confident if this is a breaking change or not to choose the right semver for this update. Please be patient, or fork and fix it to make it work for your particular project

**Comment by iamolegga** [comment]
Sorry for the long waiting, the fix was published in `3.1.3`
